pkg/daemon/ceph/osd/daemon.go

```diff
@@ -424,6 +424,13 @@ func getAvailableDevices(context *clusterd.Context, agent *OsdAgent) (*DeviceOsd
 					}
 				}
 				matchedDevice = desiredDevice
+
+				// On PVC, we must use the real name not the /mnt name since c-v calls udev
+				// udevadm will fail with
+				// stderr Unknown device, --name=, --path=, or absolute path in /dev/ or /sys expected.
+				if agent.pvcBacked {
+					matchedDevice.Name = device.RealPath
```

Comment:
Should this only apply to an encrypted volume? Or why wasn't this an issue before with PVCs?

Reply:
So before we were doing encryption we were using `/mnt/set1-data-0-7dwll` which was fine. But somehow now c-v is calling udev so this will fail since `/mnt` is not a valid system path.
So now we just use `/dev/<whatever>` for both encrypted and not encrypted. This has no side effect, we still use the same block but from a different path.

I tested a non-encrypted cluster with this code and it worked well.